network-store-server/src/main/java/com/powsybl/network/store/server/NetworkStoreRepository.java

```diff
@@ -1085,4 +1295,129 @@ public Optional<Resource<IdentifiableAttributes>> getIdentifiable(UUID networkUu
+                temporaryLimit.setSide(resultSet.getInt(5));
+                temporaryLimit.setLimitType(TemporaryLimitType.getByValue(resultSet.getString(6)));
+                temporaryLimit.setName(resultSet.getString(7));
+                temporaryLimit.setValue(resultSet.getDouble(8));
+                temporaryLimit.setAcceptableDuration(resultSet.getInt(9));
+                temporaryLimit.setFictitious(resultSet.getBoolean(10));
+
+                temporaryLimits.add(temporaryLimit);
+            }
+            return temporaryLimits;
+        }
+    }
+
+    protected <T extends LimitSelector & IdentifiableAttributes> List<TemporaryLimitAttributes> getTemporaryLimitsFromEquipments(UUID networkUuid, List<Resource<T>> resources) {
+        ArrayList<TemporaryLimitAttributes> result = new ArrayList<>();
+        if (!resources.isEmpty()) {
+            for (Resource<T> resource : resources) {
+                T equipment = resource.getAttributes();
+
+                List<TemporaryLimitAttributes> temporaryLimits = equipment.getAllTemporaryLimits();
+                temporaryLimits.forEach(e -> {
+                    e.setVariantNum(resource.getVariantNum());
+                    e.setNetworkUuid(networkUuid.toString());
+                    e.setEquipmentId(resource.getId());
+                });
+
+                result.addAll(temporaryLimits);
+            }
+        }
+        return result;
+    }
+
+    protected <T extends LimitSelector & IdentifiableAttributes> void insertTemporaryLimitsInEquipments(List<Resource<T>> equipments, List<TemporaryLimitAttributes> temporaryLimits) {
+        // Some equipments can have temporary limits.
+        // Those limits are not in the database table representation of the equipment, they are in a different
+        // table : "temporarylimit".
+        // We need to complete the equipments we get from the database by searching the corresponding temporary limits
+        // and inserting those limits inside the corresponding equipments.
+        // The choosen algorithm to do this is to retrieve all the temporary limits for a networkUuid, variantNum and
+        // side, then check each limit's equipment ID to see if it is the same ID as an equipment's.
+        // If it is, then it means the temporary limit belongs to the equipment.
+
+        if (!temporaryLimits.isEmpty() && !equipments.isEmpty()) {
+            // For each equipment, we will check if there are temporary limits with the corresponding equipment IDs.
+            // If there is, then we add the temporary limit to the equipment's temporaryLimits
+            // First, we put the temporary limits in a hashmap, with the map's key equals to the equipmentID.
+            // Then, for each equipment, we will load the corresponding temporary limits from the hashmap.
+            HashMap<String, List<TemporaryLimitAttributes>> hashTemporaryLimits = new HashMap<>(equipments.size());
```

Comment:
temporaryLimitsByEquipmentId

Reply:
Ok, renamed variable `hashTemporaryLimits` to `temporaryLimitsByEquipmentId`.